pkg/eks/eks.go

```diff
@@ -143,27 +147,65 @@ func (c *ClusterProvider) GetCredentials(cluster awseks.Cluster) error {
 	return nil
 }
 
-func (c *ClusterProvider) ListClusters() error {
+// ListClusters display details of all the EKS cluster in your account
+func (c *ClusterProvider) ListClusters(chunkSize int, output string) error {
+	// NOTE: this needs to be reworked in the future so that the functionality
+	// is combined. This require the ability to return details of all clusters
+	// in a single call.
+	printer, err := printers.NewPrinter(output)
+	if err != nil {
+		return err
+	}
+
 	if c.Spec.ClusterName != "" {
-		return c.doListCluster(&c.Spec.ClusterName)
+		if output == "table" {
+			addSummaryTableColumns(printer.(*printers.TablePrinter))
+		}
+		return c.doGetCluster(&c.Spec.ClusterName, printer)
 	}
 
-	// TODO: https://github.com/weaveworks/eksctl/issues/27
-	input := &awseks.ListClustersInput{}
-	output, err := c.Provider.EKS().ListClusters(input)
-	if err != nil {
-		return errors.Wrap(err, "listing control planes")
+	if output == "table" {
+		addListTableColumns(printer.(*printers.TablePrinter))
 	}
-	logger.Debug("clusters = %#v", output)
-	for _, clusterName := range output.Clusters {
-		if err := c.doListCluster(clusterName); err != nil {
+	return c.doListClusters(int64(chunkSize), printer)
+}
+
+func (c *ClusterProvider) doListClusters(chunkSize int64, printer printers.OutputPrinter) error {
+	allClusterNames := []*string{}
+
+	getFunc := func(chunkSize int64, nextToken string) ([]*string, *string, error) {
+		input := &awseks.ListClustersInput{MaxResults: &chunkSize}
+		if nextToken != "" {
+			input = input.SetNextToken(nextToken)
+		}
+		output, err := c.Provider.EKS().ListClusters(input)
+		if err != nil {
+			return nil, nil, errors.Wrap(err, "listing control planes")
+		}
+		return output.Clusters, output.NextToken, nil
+	}
+
+	token := ""
+	for {
+		clusters, nextToken, err := getFunc(chunkSize, token)
+		if err != nil {
 			return err
 		}
+		allClusterNames = append(allClusterNames, clusters...)
+		if nextToken != nil && *nextToken != "" {
+			token = *nextToken
+		} else {
+			break
+		}
 	}
+
+	logger.Debug("clusters = %#v", allClusterNames)
```

Comment:
This seems to print only the pointers, I'm not sure if we need it at all...
```
2018-08-02T09:10:12+01:00 [▶]  clusters = []*string{(*string)(0xc4205307e0), (*string)(0xc4205307f0), (*string)(0xc420530820), (*string)(0xc420530830)}
```

Reply:
I removed the `log' printer. It was always an after thought.